File: src/firebase.js
import firebase from 'firebase'

var firebaseConfig = {
    apiKey: "AIzaSyCHoUYE86LbyoKzP7W0t1znmLuyEKTyGmo",
    authDomain: "revolutionroboticsdev.firebaseapp.com",
    databaseURL: "https://revolutionroboticsdev.firebaseio.com",
    projectId: "revolutionroboticsdev",
    storageBucket: "revolutionroboticsdev.appspot.com",
    messagingSenderId: "54672154034",
    appId: "1:54672154034:web:97345d105e653f34"
  };
  // Initialize Firebase
  firebase.initializeApp(firebaseConfig);

  export default firebase;

File: lib/firebase.dart
import 'package:firebase/firebase.dart';

class Firebase {

  static final Firebase _singleton = Firebase._internal(); 
  
  static Database _database;
  static Storage _storage;

  factory Firebase() {
    initializeApp(
      apiKey: 'AIzaSyCHoUYE86LbyoKzP7W0t1znmLuyEKTyGmo',
      authDomain: 'revolutionroboticsdev.firebaseapp.com',
      databaseURL: 'https://revolutionroboticsdev.firebaseio.com',
      projectId: 'revolutionroboticsdev',
      storageBucket: 'revolutionroboticsdev.appspot.com'
    );

    _database = database();
    _storage = storage();

    return _singleton;
  }

  void listen(String reference, Function listener) {
    DatabaseReference ref = _database.ref(reference);
    ref.onValue.listen(listener);
  }

  void once(String reference, Function callback) {
    DatabaseReference ref = _database.ref(reference);
    ref.once('value').then((e) => callback(e));
  }

  void getImageUrl(String gsUrl, Function callback) {
    final StorageReference ref = _storage.refFromURL(gsUrl);
    ref.getDownloadURL().then((value) {
      callback(value.toString());
    });
  }

  void update(String reference, dynamic value) {
    _database.ref(reference).update(value);
  }

  Firebase._internal();
}
